src/main/java/org/spongepowered/mod/mixin/entity/vehicle/minecart/MixinEntityMinecart.java

```diff
@@ -0,0 +1,138 @@
+import net.minecraft.entity.item.EntityMinecart;
+import net.minecraft.world.World;
+import org.spongepowered.api.entity.vehicle.minecart.Minecart;
+import org.spongepowered.api.util.annotation.NonnullByDefault;
+import org.spongepowered.asm.mixin.*;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import java.io.IOException;
+
+@NonnullByDefault
+@Mixin(EntityMinecart.class)
+@Implements(@Interface(iface = Minecart.class, prefix = "minecart$"))
+public abstract class MixinEntityMinecart extends Entity {
+
+    @Shadow
+    public abstract double getDragAir();
+
+    @Shadow
+    public abstract boolean isOnGround();
+
+    private double maxSpeed = 0.4D;
+    private boolean slowWhenEmpty = true;
+    private Vector3d airborneMod = new Vector3d(0.5D, 0.5D, 0.5D);
+    private Vector3d derailedMod = new Vector3d(0.5D, 0.5D, 0.5D);
+
+    // this method overwrites the vanilla accessor for maximum speed
+    @Overwrite
+    public double func_174898_m() {
+        return this.maxSpeed;
+    }
+
+    // this method overwrite vanilla behavior to allow for a custom deceleration rate on all three axes when airborne
+    @Inject(method = "func_180459_n", at = @At(value = "FIELD", target = "net.minecraft.entity.Entity.onGround:Z", ordinal = 2))
+    public void implementCustomAirborneDeceleration(CallbackInfo ci) {
+        if (!this.isOnGround()) {
+            this.motionX /= this.getDragAir();
+            this.motionY /= this.getDragAir();
+            this.motionZ /= this.getDragAir();
+            this.motionX *= airborneMod.getX();
+            this.motionY *= airborneMod.getY();
+            this.motionZ *= airborneMod.getZ();
+        }
+    }
+
+    // this method overwrites vanilla behavior to allow for a custom deceleration rate when derailed
+    @Inject(method = "func_180459_n", at = @At(value = "INVOKE", target = "net.minecraft.entity.Entity.moveEntity(DDD)V"))
+    public void implementCustomDerailedDeceleration(CallbackInfo ci) {
+        if (this.isOnGround()) {
+            this.motionX /= 0.5D;
+            this.motionY /= 0.5D;
+            this.motionZ /= 0.5D;
+            this.motionX *= derailedMod.getX();
+            this.motionY *= derailedMod.getY();
+            this.motionZ *= derailedMod.getZ();
+        }
+    }
+
+    // this method overwrites vanilla behavior to allow the cart not to slow when empty
+    @Overwrite
+    protected void applyDrag() {
+        if (this.riddenByEntity != null || !this.slowWhenEmpty) {
+            this.motionX *= 1.0D;
+            this.motionY *= 0.0D;
+            this.motionZ *= 1.0D;
+        }
+        else {
+            this.motionX *= 0.96D;
+            this.motionY *= 0.0D;
+            this.motionZ *= 0.96D;
+        }
+    }
+
+    public MixinEntityMinecart(World worldIn) {
+        super(worldIn);
+    }
+
+    public double minecart$getMaxSpeed() {
```

Comment:
This is going to overwrite the forge method, which probably isn't want you wanted.